plugins/test/chromatic_correction_test.py

```diff
@@ -0,0 +1,147 @@
+                                          find_corresponding_points, get_chromatic_correction_dict)
+
+PIXEL_SIZE = (1e-6, 1e-6)  # m
+# Affine parameters that are used on reference image
+# Translation can be any value. Scale is fixed to one if there is some rotation and shear else both scale, shear
+# can be changed by keeping the rotation to zero
+SHEAR = 0.9
+SCALE = [2, 3]
+ROTATION = numpy.radians(0)
+TRANSLATION = [2, 1]
+# Default values when there the datasets are similar
+DEFAULT_SHEAR = 0
+DEFAULT_SCALE = [1, 1]
+DEFAULT_ROTATION = 0
+DEFAULT_TRANSLATION = [0, 0]
+
+
+class TestChromaticAberrationCorrection(unittest.TestCase):
+    """
+    Test the Chromatic Aberration Correction functions used in the plugin
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        cls.das = read_data('FOV1_green.tif')
+
+    def test_chromatic_correction_with_fake_data_compare_methods(self):
+        """Given two images in which one is reference image and other is affine transformed image of reference image.
+         The calculated affine parameters should correct the transformed image parameters back to the reference image
+         Two methods namely RSU and RSL from odemis.util.transform.py are compared to show which method yields
+         good results"""
+        # reference image
+        ref = self.das[0]
+        ref.metadata.update({model.MD_ROTATION: 0})
+        ref.metadata.update({model.MD_PIXEL_SIZE: PIXEL_SIZE})
+        ref.metadata.update({model.MD_SHEAR: 0})
+        ref.metadata.update({model.MD_POS: [0, 0]})
+        # modified reference image
+        modified_im = self.das[0]
+        modified_im.metadata.update({model.MD_ROTATION: ROTATION})
+        modified_im.metadata.update({model.MD_POS: [TRANSLATION[0] * PIXEL_SIZE[0], -TRANSLATION[1] * PIXEL_SIZE[1]]})
+        modified_im.metadata.update(
+            {model.MD_PIXEL_SIZE: [SCALE[0] * PIXEL_SIZE[0], SCALE[1] * PIXEL_SIZE[1]]})
+        modified_im.metadata.update({model.MD_SHEAR: SHEAR})
+
+        # Generate synthetic keypoints of reference and modified images based on given affine parameters
+        r_m = _rotation_matrix_from_angle(ROTATION)
+        # Make sure the given points are not in one line
+        points1 = numpy.array([[5, 5], [5, 10], [10, 8], [11, 11]])
+        shear_m = numpy.array([[1, SHEAR], [0, 1]])
+        points2 = TRANSLATION + SCALE * ((r_m @ (shear_m @ points1.transpose())).transpose())
+        print(f"points 1 {points1}")
+        print(f"points 2 {points2}")
+
+        affine = AffineTransform.from_pointset(points1, points2)
+        mat = numpy.eye(3)
+        mat[:2, :2] = affine.matrix
+        mat[:2, 2] = affine.translation
+        tx = mat[0, 2]
+        ty = mat[1, 2]
+        # Projection image y-axis is upside down, add negative sign
+        translation = [tx * PIXEL_SIZE[0], -ty * PIXEL_SIZE[1]]
+
+        i = 0
+        for image in [ref, modified_im, modified_im, modified_im]:
```

Comment:
This `for` loop does 4 completely different things for each step. => Just unroll the loop as a single code!